lib/addons/amp/index.ts

```diff
@@ -0,0 +1,131 @@
+     */
+    workspaceName?: string;
+    /** 
+     * Remote Write URL of the AMP Workspace to be used for setting up remote write.
+     */
+     prometheusRemoteWriteURL?: string;
+    /**
+     * Tag Key to passed while creating AMP workspace
+     * @default Project
+     */
+     workspaceTagKey?: string;
+    /**
+     * Tag Value to passed while creating AMP workspace
+     * @default blueprints-amp-workspace
+     */
+     workspaceTagValue?: string;
+    /**
+     * Modes supported : `deployment`, `daemonset`, `statefulSet`, and `sidecar`
+     * @default deployment
+     */
+     deploymentMode?: DeploymentMode;
+}
+
+export const enum DeploymentMode {
+    DEPLOYMENT = 'deployment',
+    DAEMONSET = 'daemonset',
+    STATEFULSET = 'statefulset',
+    SIDECAR = 'sidecar'
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps = {
+    workspaceName: 'blueprints-amp-workspace',
+    workspaceTagKey: 'Project',
+    workspaceTagValue: 'blueprints-amp-workspace',
+    deploymentMode: DeploymentMode.DEPLOYMENT
+};
+
+/**
+ * Implementation of AMP add-on for EKS Blueprints. Installs ADOT Collector.
+ */
+export class AmpAddOn implements ClusterAddOn {
+
+    readonly ampAddOnProps: AmpAddOnProps;
+
+    constructor(props?: AmpAddOnProps) {
+        this.ampAddOnProps = { ...defaultProps, ...props };
+    }
+
+    @dependable(AdotCollectorAddOn.name)
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        let finalRemoteWriteURLEndpoint: string;
+        let finalDeploymentMode: string;
+        let doc: string;
+        
+        let cfnWorkspaceProps : CfnWorkspaceProps  = {
+            alias: defaultProps.workspaceName,  
+            tags: [{
+            key: defaultProps.workspaceTagKey,
+            value: defaultProps.workspaceTagValue
+            }],
+        }
+
+        if (this.ampAddOnProps.workspaceName){
+            cfnWorkspaceProps = {
+                ...cfnWorkspaceProps,
+                alias:this.ampAddOnProps.workspaceName
+            };        
+        }
+
+        if (this.ampAddOnProps.workspaceTagKey && this.ampAddOnProps.workspaceTagValue){
+            cfnWorkspaceProps = {
+                ...cfnWorkspaceProps,
+                tags: [{
+                    key: this.ampAddOnProps.workspaceTagKey,
+                    value: this.ampAddOnProps.workspaceTagValue
+                }]
+            }
+        }
+
+        if (this.ampAddOnProps.prometheusRemoteWriteURL) {
+            finalRemoteWriteURLEndpoint = this.ampAddOnProps.prometheusRemoteWriteURL;
+        }
+        else {
+            const cfnWorkspace = new aps.CfnWorkspace(cluster.stack, 'MyAMPWorkspace', cfnWorkspaceProps);/* all optional props */ 
+            const URLEndpoint = cfnWorkspace.attrPrometheusEndpoint;
+            finalRemoteWriteURLEndpoint = URLEndpoint + 'api/v1/remote_write';
+        }
+
+        finalDeploymentMode = defaultProps.deploymentMode
```

Comment:
If you switch to the [file mode](https://github.com/aws-quickstart/cdk-eks-blueprints/pull/445/files) you will see a few lint warnings below.

Reply:
All lint issues are fixed, thankyou for pointing out.